beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -556,6 +556,45 @@ func MatchAttestations(state *pb.BeaconState, epoch uint64) (*MatchedAttestation
 	}, nil
 }
 
+// ProcessCrosslink processes crosslink and finds the crosslink
+// with enough state to make it canonical in state.
+//
+// Spec pseudocode definition:
+//	def process_crosslinks(state: BeaconState) -> None:
+//    state.previous_crosslinks = [c for c in state.current_crosslinks]
+//    previous_epoch = get_previous_epoch(state)
+//    next_epoch = get_current_epoch(state) + 1
+//    for slot in range(get_epoch_start_slot(previous_epoch), get_epoch_start_slot(next_epoch)):
+//        epoch = slot_to_epoch(slot)
+//        for crosslink_committee, shard in get_crosslink_committees_at_slot(state, slot):
+//            winning_crosslink, attesting_indices = get_winning_crosslink_and_attesting_indices(state, shard, epoch)
+//            if 3 * get_total_balance(state, attesting_indices) >= 2 * get_total_balance(state, crosslink_committee):
+//                state.current_crosslinks[shard] = winning_crosslink
+func ProcessCrosslink(state *pb.BeaconState) (*pb.BeaconState, error) {
+	state.PreviousCrosslinks = state.CurrentCrosslinks
+	prevEpoch := helpers.PrevEpoch(state)
+	nextEpoch := helpers.CurrentEpoch(state) + 1
+	for slot := helpers.StartSlot(prevEpoch); slot < helpers.StartSlot(nextEpoch); slot++ {
+		epoch := helpers.SlotToEpoch(slot)
+		committees, err := helpers.CrosslinkCommitteesAtSlot(state, slot, false /* registery change */)
+		if err != nil {
+			return nil, err
+		}
+		for _, committee := range committees {
+			crosslink, indices, err := WinningCrosslink(state, committee.Shard, epoch)
+			if err != nil {
+				return nil, err
+			}
+			attestedBalance := TotalBalance(state, indices)
+			totalBalance := TotalBalance(state, committee.Committee)
+			if 3*attestedBalance >= 2*totalBalance {
```

Comment:
What is the significance of 3?
What is the significance of 2?

These appear to be magic numbers. Please consider a named constant or inline comment to explain why 2, 3.

Reply:
Sounds, adding a comment. 
I tried to introduce 2/3 to `config.go` but it'll introduce `float64`. `2 * x > 3 * y` feels better in this case

Reply:
sg